confidential-data-hub/hub/src/config.rs

```diff
@@ -0,0 +1,404 @@
+// Copyright (c) 2024 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{env, fs, path::Path};
+
+use anyhow::*;
+use attestation_agent::config::aa_kbc_params::AaKbcParams;
+use config::{Config, File};
+use log::{debug, info};
+use serde::Deserialize;
+
+cfg_if::cfg_if! {
+    if #[cfg(feature = "ttrpc")] {
+        const DEFAULT_CDH_SOCKET_ADDR: &str = "unix:///run/confidential-containers/cdh.sock";
+    } else {
+        const DEFAULT_CDH_SOCKET_ADDR: &str = "127.0.0.1:50000";
+    }
+}
+
+#[derive(Clone, Deserialize, Debug, PartialEq)]
+pub struct KbsConfig {
+    pub name: String,
+
+    pub url: String,
+
+    pub kbs_cert: Option<String>,
+}
+
+impl KbsConfig {
+    fn new() -> Result<Self> {
+        debug!("Try to get kbc and url from env and kernel commandline.");
+        let aa_kbc_params =
+            AaKbcParams::new().context("failed to read aa_kbc_params to initialize KbsConfig")?;
+        Ok(KbsConfig {
+            name: aa_kbc_params.kbc,
+            url: aa_kbc_params.uri,
+            kbs_cert: None,
+        })
+    }
+}
+
+#[derive(Clone, Deserialize, Debug, PartialEq)]
+pub struct Credential {
+    pub resource_uri: String,
+    pub path: String,
+}
+
+#[derive(Clone, Deserialize, Debug, PartialEq)]
+pub struct ImageConfiguration {
+    /// Whether to enable image security validation
+    ///
+    /// This defaults to `false`.
+    pub security_validate: bool,
+
+    /// The flag to control whether we use registry credential file to
+    /// access the image registries.
+    ///
+    /// This defaults to `false`.
+    pub use_registry_credentials: bool,
+
+    /// Maximum number of concurrent downloads to perform during image pull.
+    ///
+    /// This defaults to [`image_rs::config::DEFAULT_MAX_CONCURRENT_DOWNLOAD`].
+    pub max_concurrent_download: usize,
```

Comment:
Let's clarify whether this refers to pulling multiple images at once, multiple layers at once, or something else.

Also we might want to say what the actual default value is and when people might want to set it higher or lower.